operator/main.go

```diff
@@ -324,17 +339,14 @@ func setupKcpWatcherReconciler(
 	if err != nil || !fileInfo.IsDir() {
 		setupLog.Error(err, "failed to read local skr chart")
 	}
-	watcherConfig := &controllers.WatcherConfig{
-		WebhookChartPath:       flagVar.skrWatcherPath,
-		SkrWebhookMemoryLimits: flagVar.skrWebhookMemoryLimits,
-		SkrWebhookCPULimits:    flagVar.skrWebhookCPULimits,
-	}
+	// set MaxConcurrentReconciles to 1 to avoid concurrent writes on
+	// the Istio virtual service resource the WatcherReconciler is managing
+	options.MaxConcurrentReconciles = 1
```

Comment:
As the watcher reconcile loop reconciles one virtual service resource (to avoid istio routing side effects when having many virtual services), we have decided to disable concurrent writes for it to avoid conflicts and due to the fact that it reconciles few watcher CRs we deemed this a suitable solution.

Reply:
What I do not understand is why you cannot concurrently work with the same virtual service cr over the API. which error would you receive and why would that not be possible concurrently? I think there is a misunderstanding on my side maybe but I think you should be able to work with a single virtual service resource from many loops at once where one will win and others should back of and retry with conflict.

Reply:
Regarding this, I think it's better just use 1 worker for watcher loop, because for each individual module operator, we have one dedicate watcher CR, so in total, we probably only have 20 watcher CRs, one worker can sufficiently handle it, and don't have to deal with concurrent write to virtual service although eventually the write will success.

Reply:
Under this argument I agree. Could we document this decision in code so that its clear what the expectation is? otherwise people will wonder why we decided on this.